streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamTransformValues.java

```diff
@@ -46,8 +58,104 @@ public KStreamTransformValuesProcessor(ValueTransformer<V, R> valueTransformer)
         }
 
         @Override
-        public void init(ProcessorContext context) {
-            valueTransformer.init(context);
+        public void init(final ProcessorContext context) {
```

Comment:
I was thinking we need to do something about the `ProcessorContext`, too. The interface is quite broad. I think what you have suggested is ok, but I'd probably like to see `register` moved elsewhere, too. At the moment the `ProcessorContext` is accommodating initialization and processing, which means we have methods on it that aren't valid in all circumstances. IMO it would be nicer to have more focussed interfaces that make it less likely that you can do something that isn't allowed. The interfaces can all be implemented by the same object, of course.